src/main/java/de/btegermany/terraplusminus/commands/TpllCommand.java

```diff
@@ -250,22 +219,63 @@ private static boolean sendPluginMessageToBungeeBridge(boolean isNextServer, Pla
         if (server != null) {
             out.writeUTF(server.getWorldName() + ", " + server.getOffset());
         } else {
-            player.sendMessage(Terraplusminus.config.getString("prefix") + "§cPlease contact server administrator. Your config is not set up correctly.");
-            return true;
+            player.sendMessage(prefix + "§cPlease contact server administrator. Your config is not set up correctly.");
+            return;
         }
-        out.writeUTF(coordinates[1] + ", " + coordinates[0]);
-        player.sendPluginMessage(Terraplusminus.instance, "bungeecord:terraplusminus", out.toByteArray());
+        out.writeUTF(lat + ", " + lon);
+        player.sendPluginMessage(Terraplusminus.instance,
+                "bungeecord:terraplusminus",
+                out.toByteArray());
 
-        player.sendMessage(Terraplusminus.config.getString("prefix") + "§cSending to another server...");
-        return true;
+        player.sendMessage(prefix + "§cSending to another server...");
     }
 
-    public boolean isDouble(String str) {
-        try {
-            Double.parseDouble(str);
-            return true;
-        } catch (NumberFormatException e) {
-            return false;
+    public static LiteralCommandNode<CommandSourceStack> create() {
+        prefix = Terraplusminus.config.getString("prefix");
+
+        return Commands.literal("tpll")
+                .then(Commands.argument(LAT_LON_HEIGHT, StringArgumentType.greedyString())
+                        .requires(TpllCommand::isPermittedDirect)
+                        .executes(TpllCommand::executeDirect))
+                .then(Commands.argument("player", ArgumentTypes.players())
+                        .then(Commands.argument(LAT_LON_HEIGHT, StringArgumentType.greedyString())
+                                .requires(TpllCommand::isPermittedTarget)
+                                .executes(TpllCommand::executeTarget))
+                        .requires(TpllCommand::isPermittedTarget)
+                        .executes(TpllCommand::sendUsageMessage))
+                .requires(TpllCommand::isPermitted)
+                .executes(TpllCommand::sendUsageMessage)
+                .build();
+    }
+
+    private static int executeTarget(@NotNull CommandContext<CommandSourceStack> ctx) throws CommandSyntaxException {
+        final PlayerSelectorArgumentResolver targetResolver = ctx.getArgument("targets", PlayerSelectorArgumentResolver.class);
+        final List<Player> targets = targetResolver.resolve(ctx.getSource());
+        final String latLonHeight = ctx.getArgument(LAT_LON_HEIGHT, String.class);
+        for (final Player target : targets) {
+            execute(ctx.getSource().getSender(), target, latLonHeight);
         }
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static boolean isPermittedTarget(@NotNull CommandSourceStack commandSourceStack) {
+        return commandSourceStack.getSender().hasPermission("t+-.forcetpll");
+    }
```

Comment:
Nitpick, non-blocking.
I find this method poorly name since it has nothing to do with the target. Something like `canSenderForceTp(...)` would probably be more appropriate.